key/benches/ed25519.rs

```diff
@@ -14,39 +14,31 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-#![feature(test)]
-
 extern crate codechain_key as ckey;
-extern crate test;
 
 use ckey::{sign, verify, Ed25519KeyPair, Generator, KeyPairTrait, Message, Random};
-use test::Bencher;
-
-#[bench]
-fn ec25519_sign(b: &mut Bencher) {
-    b.iter(|| {
-        let key_pair: Ed25519KeyPair = Random.generate().unwrap();
-        let message = Message::random();
-        let _signature = sign(message.as_ref(), key_pair.private());
-    })
-}
+use criterion::{criterion_group, criterion_main, Criterion};
 
-#[bench]
-fn ed25519_sign_and_verify(b: &mut Bencher) {
-    b.iter(|| {
-        let key_pair: Ed25519KeyPair = Random.generate().unwrap();
-        let message = Message::random();
-        let signature = sign(message.as_ref(), key_pair.private());
-        assert!(verify(&signature, message.as_ref(), key_pair.public()));
-    })
+fn ec25519_sign(c: &mut Criterion) {
+    c.bench_function("ec25519_sign", |b| {
+        b.iter(|| {
+            let key_pair: Ed25519KeyPair = Random.generate().unwrap();
+            let message = Message::random();
+            let _signature = sign(message.as_ref(), key_pair.private());
```

Comment:
Why did you remove the line below?
`         assert!(verify(&signature, message.as_ref(), key_pair.public())); `

Reply:
It is from `ec25519_sign`, not `ed25519_sign_and_verify`. There was no assert in original code.
I split the commit to make it clear